packages/service-library/src/servicelib/tracing.py

```diff
@@ -0,0 +1,33 @@
+from typing import TypeAlias
+
+from opentelemetry import context as otcontext
+from opentelemetry import trace
+from opentelemetry.instrumentation.logging import LoggingInstrumentor
+from settings_library.tracing import TracingSettings
+
+TracingContext: TypeAlias = otcontext.Context | None
+
+
+def _is_tracing() -> bool:
+    return trace.get_current_span().is_recording()
+
+
+def get_context() -> TracingContext:
+    if not _is_tracing():
+        return None
+    return otcontext.get_current()
+
+
+def attach_context(context: TracingContext) -> None:
```

Comment:
in `get_context` you expliclty return `None` and here not. :-)

Careful with annotations! 

```python
def get_context() -> TracingContext | None:
    if _is_tracing():
        return otcontext.get_current()

def attach_context(context: TracingContext | None) -> None:
      if context:
           ocontext.attach(context)
```

Reply:
Yep, explained in the answer to the other remark